common/beerocks/bcl/unit_tests/wireless_utils_test.cpp

```diff
@@ -13,120 +13,265 @@
+    }
+
+    if (bandwidth >= beerocks::eWiFiBandwidth::BANDWIDTH_80) {
+        return center_channel - (first_vht_80_mhz_channel - first_vht_20_mhz_channel);
+    }
+
+    return center_channel;
 }
 
-// See https://docs.google.com/spreadsheets/d/1J2IYuGjFX_OQQpgb4OgsyDx73klAUk6qSmeB4acJ5us
-// for the table we need to conform to.
-TEST(wireless_utils_test, get_operating_class_by_channel_should_return_valid)
+/**
+ * @brief Gets a list of all possible valid parameters to get_operating_class_by_channel() function
+ * together with the expected result for each one of them.
+ *
+ * Creates a container with all possible combinations of channel number and channel bandwidth to
+ * create a valid sWifiChannel to use as parameter to get_operating_class_by_channel() function and
+ * the expected operating class to be obtained as result.
+ *
+ * @return vector with pairs of valid WiFi channels and their expected operating class result.
+ */
+std::vector<tGetOperatingClassByChannelParam> get_operating_class_by_channel_valid_parameters()
 {
-    EXPECT_EQ(get_operating_class_by_channel_helper(1, 20), 81);
-    EXPECT_EQ(get_operating_class_by_channel_helper(13, 20), 81);
+    std::vector<tGetOperatingClassByChannelParam> result;
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(14, 20), 82);
+    for (const auto &entry : table_E_4_global_operating_classes_802_11) {
+        for (const auto channel : entry.channels) {
+            beerocks::message::sWifiChannel ch;
+            ch.channel           = get_starting_channel(channel, entry.bandwidth);
+            ch.channel_bandwidth = entry.bandwidth;
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(1, 40), 83);
-    EXPECT_EQ(get_operating_class_by_channel_helper(9, 40), 83);
+            result.emplace_back(std::make_tuple(ch, entry.operating_class));
+        }
+    }
 
-    /* Note: we have no way to distinguish 40- and 40+ */
-    EXPECT_EQ(get_operating_class_by_channel_helper(10, 40), 84);
-    EXPECT_EQ(get_operating_class_by_channel_helper(13, 40), 84);
+    return result;
+}
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(36, 20), 115);
-    EXPECT_EQ(get_operating_class_by_channel_helper(48, 20), 115);
+/**
+ * @brief Gets a list of invalid parameters to get_operating_class_by_channel() function.
+ *
+ * Creates a container of sWifiChannel objects built up with invalid combinations of channel
+ * number and channel bandwidth. The get_operating_class_by_channel() function returns 0 when
+ * called with any of these invalid channels as parameter.
+ *
+ * @return vector of invalid WiFi channels.
+ */
+std::vector<tGetOperatingClassByChannelParam> get_operating_class_by_channel_invalid_parameters()
+{
+    // clang-format off
+    const beerocks::eWiFiBandwidth bandwidths[]{
+        beerocks::BANDWIDTH_UNKNOWN,
+		beerocks::BANDWIDTH_20,
+        beerocks::BANDWIDTH_40,
+		beerocks::BANDWIDTH_80,
+        beerocks::BANDWIDTH_80_80,
+		beerocks::BANDWIDTH_160
```

Comment:
This indentation is also weird. I guess intentional, but I don't understand why...

Reply:
Looks like tabs vs spaces, and github uses a different tab size than Mario's editor is configured for.

Reply:
Sorry, my bad. You are right: I recently changed my editor and I didn't realize it was configured to use tabs instead of spaces by default. 
Maybe clang-format should take care of this and replace tabs with 4 spaces. 

Fixed